nssp_demo/forecast_non_target_visits.R

```diff
@@ -0,0 +1,116 @@
+script_packages <- c(
+  "dplyr",
+  "tidyr",
+  "tibble",
+  "readr",
+  "stringr",
+  "fs",
+  "fable",
+  "jsonlite",
+  "argparser",
+  "arrow"
+)
+
+## load in packages without messages
+for (package in script_packages) {
+  suppressPackageStartupMessages(
+    library(package,
+      character.only = TRUE
+    )
+  )
+}
+
+
+fit_and_forecast <- function(other_data,
+                             n_forecast_days = 28,
+                             n_samples = 2000) {
+  forecast_horizon <- glue::glue("{n_forecast_days} days")
+
+  fit <-
+    other_data |>
+    filter(data_type == "train") |>
+    model(
+      comb_model = combination_ensemble(
+        ETS(log(ed_visits) ~ trend(method = c("N", "M", "A"))),
+        ARIMA(log(ed_visits))
+      ),
+      arima = ARIMA(log(ed_visits)),
+      ets = ETS(log(ed_visits) ~ trend(method = c("N", "M", "A")))
+    )
+
+  forecast_samples <- fit |>
+    generate(h = forecast_horizon, times = n_samples) |>
+    as_tibble() |>
+    mutate(ed_visits = .sim, .draw = as.integer(.rep)) |>
+    filter(.model == "comb_model") |>
+    select(date, .draw, other_ed_visits = ed_visits)
+
+  forecast_samples
+}
+
+main <- function(model_dir, n_forecast_days = 28, n_samples = 2000) {
+  # to do: do this with json data that has dates
+  data_path <- path(model_dir, "data", ext = "csv")
+
+  other_data <- read_csv(
+    data_path,
+    col_types = cols(
+      disease = col_character(),
+      data_type = col_character(),
+      ed_visits = col_double(),
+      date = col_date()
+    )
+  ) |>
+    mutate(disease = if_else(
+      disease == disease_name_nssp,
+      "Disease", disease
+    )) |>
+    pivot_wider(names_from = disease, values_from = ed_visits) |>
+    mutate(Other = Total - Disease) |>
+    select(date, ed_visits = Other, data_type) |>
+    as_tsibble(index = date)
+
+  forecast_samples <- fit_and_forecast(other_data, n_forecast_days, n_samples)
+
+  save_path <- path(model_dir, "other_ed_visits_forecast", ext = "parquet")
+  write_parquet(forecast_samples, save_path)
+}
+
+
+p <- arg_parser(
+  "Forecast other (non-target-disease) ED visits"
+) |>
+  add_argument(
+    "--model-dir",
+    help = "Directory containing the model data",
+  ) |>
+  add_argument(
+    "--n-forecast-days",
+    help = "Number of days to forecast",
+    default = 28L
+  ) |>
+  add_argument(
+    "--n-samples",
+    help = "Number of samples to generate",
+    default = 2000L
+  )
+
+argv <- parse_args(p)
+model_dir <- path(argv$model_dir)
+n_forecast_days <- argv$n_forecast_days
+n_samples <- argv$n_samples
```

Comment:
I just learned about [`list2env`](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/list2env), which does this.

Reply:
Maybe I'm not understanding what you're suggesting, but I don't think the readability /  safety versus concision tradeoff is in its favor with only three args

Reply:
That's a reasonable conclusion.